doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+.. note::
+
+    The `ref_name` needs to be above the section you want to reference with the
+    following syntax
+
+    ::
+
+        .. _ref_name:
+
+        secton_header
+        --------------
+
+Special links
+^^^^^^^^^^^^^
+
+.. attention::
+
+  The items in this section are still in development and have not been incorporated into nest:master!
+
+
+To link PyNEST API functions used in-text to the API reference page use the following syntax:
+
+::
+
+   :py:func:`.Create`
+
+
+Rendered as
+
+   :py:func:`.Create`
+
+
+To link terms to the glossary page use the HoverXTooltip role from Mahdi Enan (INM-6)
+
+::
+
+  :hxt_ref:`E_L`
+
+To link to external projects (PyNN, Elephant, nestml), you can treat references as you would in your local project
+with the addition of the intersphinx unique identifer (see also conf.py >  intersphinx_mapping).
+
+::
+
+  :doc:`tuturial for nestml <nestml:tutorials>`
+  :py:func:`pyNN.utility.get_simulator`
+
+Bibliography style
+------------------
+
+The reStructuredText reference style is used throughout documentation so links
+are autogenerated and a consistent format is used.
+
+For in-text citations, we use the reStructuredText numeric style ``[1]_``.
+
+
+For example:
+
+.. code-block:: none
+
+    The following example is based on Smith [1]_.
+    [2]_ contains a technically detailed description.
+
+Please ensure your reference follows the following guidelines:
+
+* References with more than 5 authors use 'et al.'.
+* Use initials for first name of authors
+* Surname precedes first name for all authors
+* No comma follows surname
+* Full stop after every section of bibliography.
+* No formatting such as italics, bold, or underline.
+* Full title of journal
+* Article titles written in sentence case
+* Year follows author(s), in parentheses
+* Volume, can be optionally followed by issue in parentheses,  a colon
+  separates volume and page range.
+* Include a linked DOI, if available
+
+.. code-block:: none
+
+ References
+ -----------
+
+ .. [1] Smith J. and Jones M (2009). Title of cool paper. Journal of
+        Awesomeness. 3:7-29. <DOI>
+
+ .. [2] Sander M., et al (2011). Biology of the sauropod dinosaurs: the
+        evolution of gigantism. Biological Reviews. 86(1):117-155.
+        https://doi.org/10.1111/j.1469-185X.2010.00137.x
+
+How we write
+------------
+
+Headings and subheadings
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Headings and subheadings should describe the purpose of the section.
+
+Begin with a descriptive verb or begin with `How to ...`
+
+Headings should explain the section in a short phrase.
+
+Use the verb stem and not the gerund ('ing') form of verbs. Not 'Adding a
+model', but  'Add a model'.
+
+Avoid section names like `Introduction` or `Part 1`.
+
+One word subheadings are acceptable, if the section is short and the meaning is clear.
+
+Use sentence case for headings and subheadings, i.e., begin with an uppercase
+letter but with all other words in lower case (except proper nouns).
```

Comment:
I wonder if we should enforce this in displaying by using a CSS `text-transform` on the resulting `<hX>`tags. Or would you rather enforce this by having a linting check during the static code check phase in the CI?

Reply:
There will always be exceptions, e.g. when `NEST` is part of a heading ...